addOns/exim/src/main/java/org/zaproxy/addon/exim/pcap/PcapUtils.java

```diff
@@ -0,0 +1,231 @@
+    private static boolean isHttpStream(TcpStream stream) {
+        List<TCPPacket> packets = stream.getPackets();
+        for (TCPPacket packet : packets) {
+            if (packet.getPayload() != null && !packet.getPayload().isEmpty()) {
+                try {
+                    HttpRequestHeader reqHeader =
+                            new HttpRequestHeader(packet.getPayload().toString());
+                    return true;
+                } catch (HttpMalformedHeaderException e) {
+                    return false;
+                }
+            }
+        }
+        return false;
+    }
+
+    // TODO: Implement robust http flow reconstruction
+    // For now we ignore retransmission/missing/overlapping/out-of-order data in tcp segments.
+    public static String getHttpRequestFlow(TcpStream stream) {
+        List<TCPPacket> tcpPackets = stream.getPackets();
+        StringBuilder requestFlow = new StringBuilder();
+        TransportStreamId requestDirection =
+                new TransportStreamId(tcpPackets.get(0)); // client is opening the tcp connection
+        for (TCPPacket tcpPacket : tcpPackets) {
+            if (tcpPacket.getPayload() != null
+                    && requestDirection.equals(new TransportStreamId(tcpPacket))) {
+                requestFlow.append(
+                        tcpPacket
+                                .getPayload()
+                                .toString()); // naively reassemble the client requests
+            }
+        }
+        return requestFlow.toString();
+    }
+
+    public static byte[] getHttpResponseFlow(TcpStream stream) {
+        List<TCPPacket> tcpPackets = stream.getPackets();
+        ByteArrayOutputStream responseFlow = new ByteArrayOutputStream();
+        TransportStreamId responseDirection =
+                new TransportStreamId(tcpPackets.get(0))
+                        .oppositeFlowDirection(); // server is receiving connection
+        for (TCPPacket tcpPacket : tcpPackets) {
+            if (tcpPacket.getPayload() != null
+                    && responseDirection.equals(new TransportStreamId(tcpPacket))) {
+                try {
+                    responseFlow.write(tcpPacket.getPayload().getArray());
+                } catch (IOException e) {
+                    LOGGER.error("Failed to write response payload: " + e.getMessage());
+                }
+            }
+        }
+        return responseFlow.toByteArray();
+    }
+
+    public static List<HttpMessage> constructHttpMessages(String requestFlow, byte[] responseFlow) {
+        List<HttpMessage> requestsMessages = constructHttpRequests(requestFlow);
+        return constructHttpResponses(responseFlow, requestsMessages);
+    }
+
+    private static List<HttpMessage> constructHttpRequests(String requestFlow) {
+        List<HttpMessage> requests = new ArrayList<>();
+        ArrayList<String> flow = new ArrayList<>(List.of(requestFlow.split("\r\n\r\n")));
```

Comment:
I think we have a vote constant for this `HttpHeader.CRLF` if I recall correctly

Reply:
`HttpHeader.CRLF` is only once '\r\n', so I redefined above a `DOUBLE_CRLF` constant based on this one above